### ISSUE TITLE
$query->expr()->in and notIn function does not support empty array

### ISSUE THREAD

**Issue by VincentLanglet** [doctrine/orm#8033]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.7.x

#### Summary

When writing
```
$query->andWhere($query->expr()->in("$alias.foo", $bar));
```

I get an error if the $bar array is an empty array.

#### Current behavior

```
$query->andWhere($query->expr()->in("$alias.foo", $bar));
```
Throw an exception.

But 
```
$query->andWhere("$alias.foo IN (:bar)");
$query->setParameter('bar', []);
```
Works.

So this should (easily ?) be fixable.

#### Expected behavior

No error

**Comment by SenseException** [comment]
What SQL do you expect on an empty array?

**Comment by VincentLanglet** [comment]
@SenseException 

When writing
```
$query->andWhere("$alias.foo IN (:bar)");
$query->setParameter('bar', []);
```
I get no row in database

So I expect that
```
$query->expr()->in("$alias.foo", [])
```
Return some SQL giving the same result.

`$alias.foo IN (NULL)` could do the job.

When writing
```
$query->expr()->notIn("$alias.foo", [])
```
I would have expected to return every row (or every row with `foo` not null). But it seems that
```
$query->andWhere("$alias.foo NOT IN (:bar)");
$query->setParameter('bar', []);
```
Return no row.

So in this case `$alias.foo NOT IN (NULL)` could do the job again.

The  `NOT IN` behaviour is weird for `$query->andWhere("$alias.foo NOT IN (:bar)")->setParameter('bar', [])`. It maybe can be improved (and then, `NOT IN (NULL)` would not be a solution) but it's another question.

**Comment by VincentLanglet** [comment]
@SenseException After a lot of investigation, I found that when you write
```
$query->andWhere("$alias.foo IN (:bar)");
$query->setParameter('bar', []);
```
Or
```
$query->andWhere("$alias.foo NOT IN (:bar)");
$query->setParameter('bar', []);
```

The params are processed by the method `expandListParameters` of the class `Doctrine\DBAL\SQLParserUtils` by the following line.
$expandStr = $count > 0 ? implode(', ', array_fill(0, $count, '?')) : 'NULL';

Which means that the executed queries are `IN (NULL)` and `NOT IN (NULL)` ; so I expect the same for the `$query->expr()`.

I made a PR.
